### PR TITLE
Docker files and such.

### PR DESCRIPTION
**Pull request by jcordial**
So, you need to have docker installed, but provided you don't have a server already running on 80 you should be able to run `docker-compose up` in this directory. Eventually you should get a message:
```
info: chance api has started on port: 4000
```
Port 4000 on the container running the node app is being forwarded to port 80 on the host machine, so you should be able to visit [http://localhost](localhost) and get a screen that looks like:
screen shot 2018-08-16 at 6 44 55 pm

The project is using `nodemon` so it should restart as changes are detected in the code. Let me know if you have any questions. I'll probably be in or at least available tomorrow afternoon after the IWiT/Eureka volunteer stuff in the morning.

With this method, instead of using `npm install`, users need to instead add the dependency to the `package.json` file and restart the app. Or, you could run:

```
docker exec -it chance_node_1 npm install {package}
```

This is because the actual node application is running in docker/linux, so any packages with native extensions need those native extensions built for linux.

( It's `chance` here because that's the name of the project folder. If you renamed the folder when you checked out the project, the name might be different... so you see why maybe it's best/easiest just to say "add packages to the `packages.json` file and restart. )

### CODE REVIEW

**Comment by jcordial** [comment]
As an example of how to connect to Postgres in the application:
```
postgres: {
    client: 'pg',
    connection: {
      host: process.env.POSTGRES_HOST,
      user: process.env.POSTGRES_USER,
      password: process.env.POSTGRES_PASSWORD,
      database: process.env.POSTGRES_DB
    },
    pool: { min: 2, max: 10 },
    migrations: { tableName: 'knex_migrations' }
  }
```

**Comment by jcordial** [comment]
@EPICmynamesBG @rcarroll380 It's all configurable. If the end user needs to change the port bindings, they can update the `docker-compose.override.yaml` file to: 
```yaml
version: '3'
services:
  node:
    ports:
      - 8080:4000
    environment:
      NODE_ENV: development
      DEBUG: "express:*"
    command:
    - /bin/sh
    - -c
    - |
      yarn
      yarn prestart
      yarn start:debug
```
The only lines added are the `ports` config, which configures the `node` container to forward port 4000 to port 8080 on the host machine. I made no presumptions about which port the end developer has open on their machine given at any particular time I have things running on 80, 8000, and 8080, amongst others.

File: package.json [hunk 1]
```diff
@@ -26,14 +38,21 @@
   "homepage": "https://github.com/zylo/chance#readme",
   "dependencies": {
     "body-parser": "^1.18.2",
+    "chai": "^4.1.2",
+    "chai-http": "^4.0.0",
```
Comment (nguyenct) [inline package.json]: Wondering why we add `chai` and `chai-http`. Should let them choose their own assertion lib

File: package.json [hunk 2]
```diff
@@ -26,14 +38,21 @@
...
     "express": "^4.16.3",
     "helmet": "^3.12.0",
     "js-yaml": "^3.11.0",
     "json-refs": "^3.0.4",
+    "knex": "^0.14.6",
     "lodash": "^4.17.5",
     "moment": "^2.22.0",
+    "pg": "^7.4.1",
+    "sqlite3": "^4.0.0",
+    "swagger": "^0.7.5",
```
Comment (nguyenct) [inline package.json]: I don't think this is required. Looks like an old package. We don't use this in waterworks.

**Comment by rcarroll380** [comment]
@EPICmynamesBG or @nguyenct or @jcordial  can you make these changes today? I want to send this off to a candidate by the end of the day.

**Comment by nguyenct** [comment]
I'm on it

File: bin/wait.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const knex = require('knex');
+const { defaultConnection } = require('../config/config');
+const logger = require('../api/helpers/logger');
+
+const wait = async (times = 0) => {
+  if (times > 40) {
+    logger.error('Could not connect to postgress. Please make sure database is running');
```
Comment (EPICmynamesBG) [inline bin/wait.js]: two `s` ?

**Pull request merged**